src/internal/components/configurationControls/axis/NamePositionControl.tsx

```diff
@@ -0,0 +1,56 @@
+// (C) 2019 GoodData Corporation
+import * as React from "react";
+import { InjectedIntlProps, injectIntl } from "react-intl";
+import get = require("lodash/get");
+
+import DropdownControl from "../DropdownControl";
+import { getTranslatedDropdownItems } from "../../../utils/translations";
+import { xAxisPositionDropdownItems, yAxisPositionDropdownItems } from "../../../constants/dropdowns";
+import { IConfigItemSubsection } from "../../../interfaces/ConfigurationPanel";
+import { IVisualizationProperties } from "../../../interfaces/Visualization";
+
+class NamePositionControl extends React.PureComponent<IConfigItemSubsection & InjectedIntlProps, {}> {
+    public render() {
+        const { axisVisible, axisNameVisible, namePosition } = this.getControlProperties();
+        const { axis, properties, pushData, disabled, configPanelDisabled, intl } = this.props;
+
+        const isDisabled = disabled || !axisVisible || !axisNameVisible;
+        const items = getTranslatedDropdownItems(
+            this.isXAxis() ? xAxisPositionDropdownItems : yAxisPositionDropdownItems,
+            intl,
+        );
+        return (
+            <DropdownControl
+                value={namePosition}
+                valuePath={`${axis}.name.position`}
+                labelText="properties.axis.name.position"
+                disabled={isDisabled}
+                showDisabledMessage={!configPanelDisabled && isDisabled}
+                properties={properties}
+                pushData={pushData}
+                items={items}
+            />
+        );
+    }
+
+    private isXAxis(): boolean {
+        return this.props.axis === "xaxis" || this.props.axis === "secondary_xaxis";
+    }
+
+    private getControlProperties(): IVisualizationProperties {
+        const axis: string = this.props.axis;
+        const controls = get(this.props, "properties.controls", {});
```

Comment:
how about 
`const controlsAxis = get(this.props, "properties.controls.${axis}", {});`?

Reply:
Fixed